tools/run_tests/xds_k8s_test_driver/framework/helpers/grpc.py

```diff
@@ -35,3 +42,106 @@ def status_eq(grpc_status_int: int, grpc_status: grpc.StatusCode) -> bool:
 def status_pretty(grpc_status: grpc.StatusCode) -> str:
     """Formats the status code as (int, NAME), f.e. (4, DEADLINE_EXCEEDED)"""
     return f"({grpc_status.value[0]}, {grpc_status.name})"
+
+
+@dataclasses.dataclass(frozen=True)
+class PrettyStatsPerMethod:
+    method: str
+    rpcs_started: int
+    result: Dict[str, int]
+
+    @functools.cached_property  # pylint: disable=no-member
+    def total_count(self):
+        return sum(self.result.values())
+
+    @staticmethod
+    def from_response(
+            method_name: str,
+            method_stats: grpc_testing.MethodStats) -> "PrettyStatsPerMethod":
+        stats: Dict[str, int] = dict()
+        for status_int, count in method_stats.result.items():
+            status: Optional[grpc.StatusCode] = status_from_int(status_int)
+            status_formatted = status_pretty(status) if status else "None"
+            stats[status_formatted] = count
+        return PrettyStatsPerMethod(
+            method=method_name,
+            rpcs_started=method_stats.rpcs_started,
+            result=stats,
+        )
+
+
+def accumulated_stats_pretty(
+    accumulated_stats: grpc_testing.LoadBalancerAccumulatedStatsResponse,
+    *,
+    ignore_empty: bool = False,
+) -> str:
+    """Pretty print LoadBalancerAccumulatedStatsResponse.
+
+    Example:
+      - method: EMPTY
+        rpcs_started: 0
+        result:
+          (2, UNKNOWN): 20
+      - method: UNARY
+        rpcs_started: 31
+        result:
+          (0, OK): 10
+          (14, UNAVAILABLE): 20
+    """
+    # Only look at stats_per_method, as the other fields are deprecated.
+    result: List[Dict] = []
+    for method_name, method_stats in accumulated_stats.stats_per_method.items():
+        pretty_stats = PrettyStatsPerMethod.from_response(
+            method_name, method_stats)
+        # Skip methods with no RPCs reported when ignore_empty is True.
+        if ignore_empty and not max(pretty_stats.rpcs_started,
+                                    pretty_stats.total_count):
+            continue
+        result.append(dataclasses.asdict(pretty_stats))
+
+    return yaml.dump(result, sort_keys=False)
+
+
+@dataclasses.dataclass(frozen=True)
+class PrettyLoadBalancerStats:
+    num_failures: int
+    rpcs_by_peer: "RpcsByPeer"
```

Comment:
RpcsByPeer here is a dict with peer as key and the number of rpcs from that peer as value right? Can we mention that somewhere?

Reply:
Good point.